replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,335 @@
+    final Metadata metadata = new MetadataImpl(((Map) info.getData()).get("raw-metadata"),
+        info.getDataClass(), task.getId(),
+        Date.from(info.getLastModified()));
+
+    final OperationType type = task.getOperation();
+    switch (type) {
+      case HARVEST:
+      case REPLICATE_FROM:
+        doCreate(task, metadata, destination, local);
+        break;
+      case REPLICATE_TO:
+        doCreate(task, metadata, local, destination);
+        break;
+      case DELETE:
+        doDelete(task, metadata, destination);
+        break;
+      default:
+        // should never hit this since deserialization will return UNKNOWN
+        task.unlock();
+    }
+  }
+
+  @Nullable
+  private NodeAdapter getDestinationAdapter(Task task) {
+    Site site;
+    final String siteId = task.getQueue().getSite();
+    try {
+      site = siteManager.get(siteId);
+    } catch (NotFoundException e) {
+      LOGGER.debug("Unable to find site {} for queue", siteId, e);
+      return null;
+    } catch (ReplicationPersistenceException e) {
+      LOGGER.debug("Failed to fetch site {} for queue", siteId, e);
+      return null;
+    }
+
+    return nodeAdapters.factoryFor(site.getType())
+        .create(site.getUrl());
+  }
+
+  private void doCreate(final Task task, final Metadata metadata, final NodeAdapter source,
+      final NodeAdapter destination)
+      throws InterruptedException {
+    final String metadataId = task.getId();
+    final String operation = task.getOperation().name();
+
+    try {
+      // TODO: remove existence check once better reporting is done by adapters
+      // see https://github.com/connexta/replication/issues/235
+      if (source.exists(metadata)) {
+        CreateRequest request = new CreateRequestImpl(List.of(metadata));
+        boolean success = destination.createRequest(request);
+        if (!success) {
+          task.fail(ErrorCode.OPERATION_FAILURE,
+              String.format("Failed to %s metadata %s from %s to %s",
+                  operation,
+                  metadataId,
+                  source.getSystemName(),
+                  destination.getSystemName()));
+          return;
+        }
+      } else {
+        LOGGER.info("Received task to create metadata on site {}, but it was deleted on {}",
+            destination.getSystemName(), source.getSystemName());
+        task.fail(ErrorCode.NO_LONGER_EXISTS, String
+            .format("Metadata deleted from site %s before able to create", source.getSystemName()));
+        return;
+      }
+    } catch (AdapterException ae) {
+      LOGGER.debug("Failed to retrieve resource", ae);
+      task.fail(ErrorCode.OPERATION_FAILURE,
+          String.format("Failed to read resource from %s", destination.getSystemName()));
+      return;
+    }
+
+    URI uri = task.getResource().flatMap(ResourceInfo::getResourceUri).orElse(null);
+    if (uri != null) {
+      ResourceRequest request = new ResourceRequestImpl(metadata);
```

Comment:
This is the only request we should support right now when dealing with Ion and existing DDFs.